komodo/snyk_reporting.py

```diff
@@ -0,0 +1,158 @@
+        ids.add(issue.id)
+    return result
+
+
+def find_vulnerabilities(
+    releases: Dict[str, Dict[str, str]],
+    repository: Dict[str, Any],
+    org: OrganizationManager,
+) -> Dict[str, List[Vulnerability]]:
+    result = dict()
+
+    for release_name, packages in releases.items():
+        pip_packages = filter_pip_packages(packages=packages, repository=repository)
+        snyk_search_string = create_snyk_search_string(pip_packages)
+        snyk_result = org.test_pipfile(snyk_search_string)
+        vulnerability_issues = get_unique_issues(snyk_result.issues.vulnerabilities)
+        result[release_name] = vulnerability_issues
+
+    return result
+
+
+def format_console(vulnerabilities: Dict[str, List[Vulnerability]]) -> str:
+    result = "Security Vulnerabilities:\n"
+    for release, vulns in vulnerabilities.items():
+        result += release + "\n"
+        if not vulns:
+            result += "No security vulnerabilities found!\n"
+            continue
+        for vul in vulns:
+            result += _CONSOLE_VULNERABILITY_FORMAT.format(
+                id=vul.id,
+                package=vul.package,
+                version=vul.version,
+                title=vul.title,
+                url=vul.url,
+                identifiers=vul.identifiers,
+                severity=vul.severity,
+            )
+        result += "\n"
+    return result
+
+
+def format_github(vulnerabilities: Dict[str, List[Vulnerability]]) -> str:
+    result = "Snyk Security Report\n==\n\n"
+    for release, vulns in vulnerabilities.items():
+        result += f"{release}\n--\n"
+        if not vulns:
+            result += "No security vulnerabilities found!\n"
+            continue
+        result += "| id | Package | Version | Title | URL | Identifiers | Severity |\n| - | - | - | - | - | - | - |\n"
+        for vul in vulns:
+            result += _GITHUB_VULNERABILITY_FORMAT.format(
+                id=vul.id,
+                package=vul.package,
+                version=vul.version,
+                title=vul.title,
+                url=vul.url,
+                identifiers=vul.identifiers,
+                severity=vul.severity,
+            )
+        result += "\n"
+    return html.escape(result)
+
+
+def snyk_main(args_in: Dict[str, str], api_token: str) -> None:
+    if api_token is None:
+        raise ValueError(
+            "No api token given, please set the environment variable SNYK_API_TOKEN."
+        )
+    args = parse_args(args=args_in)
+    client = SnykClient(api_token)
+    org = client.organizations.get(args.orgid)
+
+    releases = args.release_folder if args.release_folder else args.release
+
+    vulnerabilities = find_vulnerabilities(
+        releases=releases, repository=args.repo, org=org
+    )
+
+    if args.format_github:
+        print(format_github(vulnerabilities=vulnerabilities))
+    else:
+        print(format_console(vulnerabilities=vulnerabilities))
+
+
+if __name__ == "__main__":
```

Comment:
Is this necessary when it is installed as a console script?

Reply:
No, but it eases testing slightly since you can run it in your IDE without installing. Also did it since it seems to be convention in these files, but nothing against removing it.

Reply:
No problem 👍 Feel free to keep it...